lib/middleware/karma.js

```diff
@@ -206,6 +204,9 @@ function createKarmaMiddleware (
               scriptTags.push(`<script type="${scriptType}" src="${filePath}" ${crossOriginAttribute}></script>`)
             }
           }
+          for (const script of scriptTags) {
+            scriptUrls.push(script.replace(/</g, '\\x3C').replace(/>/g, '\\x3E'))
```

Comment:
Why is this suddenly needed?  We had '<link' for a long time no problems.

Reply:
scriptUrls (and %SCRIPT_URL_ARRAY%) are both not used in normal Karma default option (iframe). This variable is only used in the no-iframe in parent mode option. '<link' logic is never implemented properly in no-iframe in parent mode. For my use case, all my html files are preprocessed to .js files and so we only need to construct '<script>' element. I find this bug until we need to load .css files through '<link'.

Reply:
Please prefix the for loop with a comment something like:
```
// For client_with_context only, prepare for DOM parser
```
(Mixing these completely different options in the same code is not a good design. We should break this function into parts common to all forms and the parts special to each form and call them in the if/then branches above).

Reply:
Done.